### PR TITLE
Update vttjs

### PR DESCRIPTION
**Pull request by gkatsev**
Fixes https://github.com/videojs/video.js/issues/4093

### CODE REVIEW

**Comment by justinanastos** [comment]
@gkatsev I find that this fixes the issue I'd seen previously in #4093. I see this is targeted at `master`, will this be applied to the 5.x branch as well?

**Comment by gkatsev** [comment]
Yup, I'll make another PR against `5.x` when this is complete. Thanks for verifying that it works for you!

**Comment by justinanastos** [comment]
You're welcome @gkatsev !

File: src/js/tracks/text-track.js [hunk 1]
```diff
@@ -336,7 +336,17 @@ class TextTrack extends Track {
    * @param {TextTrack~Cue} cue
    *        The cue to add to our internal list
    */
-  addCue(cue) {
+  addCue(_cue) {
+    let cue = _cue;
+
+    if (!(_cue instanceof window.vttjs.VTTCue)) {
+      cue = new window.vttjs.VTTCue(_cue.startTime, _cue.endTime, _cue.text);
```
Comment (BrandonOCasey) [inline src/js/tracks/text-track.js]: would it be possible/better to expose this on videojs rather than window?

File: src/js/tracks/text-track.js [hunk 2]
```diff
@@ -336,7 +336,17 @@ class TextTrack extends Track {
    * @param {TextTrack~Cue} cue
    *        The cue to add to our internal list
    */
-  addCue(cue) {
+  addCue(_cue) {
+    let cue = _cue;
```
Comment (BrandonOCasey) [inline src/js/tracks/text-track.js]: can we change this to `cueCopy` and change `_cue` back to just `cue`?

Comment (gkatsev) [inline src/js/tracks/text-track.js]: We could. I just thought it would be better to leave the rest of the code as is and keep the change relatively self-contained.

File: src/js/tracks/text-track.js (hunk 1 above)
Comment (gkatsev) [inline src/js/tracks/text-track.js]: it's more or less exposed globally right now plus we've expected it to be exposed. For the 5.x version, we definitely should keep it exposed on window because it'll be easier to make it backwards compat.
For this PR, we could look into making a version of vttjs that doesn't bind itself to `window`. Though, I think it should be done as a separate PR rather than part of this work.

Comment (misteroneill) [inline src/js/tracks/text-track.js]: My feeling here is that vttjs is kind of its own thing - we just use it and it should stand alone more or less.

Comment (gkatsev) [inline src/js/tracks/text-track.js]: @misteroneill sort of. Given that we're using a custom fork, we can do what we want with it at this point.

The only thing I can imagine that would be tricky to get right to have vttjs not export itself globally is getting the novtt build working correctly.

File: src/js/tracks/text-track.js (hunk 2 above)
Comment (BrandonOCasey) [inline src/js/tracks/text-track.js]: I think that it's more a self documenting code change that would be worth doing, as `_cue` doesn't really have any significance to tell you what it actually is.

File: src/js/tracks/text-track.js (hunk 1 above)
Comment (BrandonOCasey) [inline src/js/tracks/text-track.js]: Im open to doing it in another PR

File: src/js/tracks/text-track.js (hunk 2 above)
Comment (gkatsev) [inline src/js/tracks/text-track.js]: What about naming it `originalCue`?

Comment (BrandonOCasey) [inline src/js/tracks/text-track.js]: sure that works too
